frontend/packages/ceph-storage-plugin/src/components/independent-dashboard-page/details-card/card.tsx

```diff
@@ -0,0 +1,103 @@
+import { SubscriptionModel } from '@console/operator-lifecycle-manager';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { getName, getInfrastructurePlatform } from '@console/shared';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import DetailItem from '@console/shared/src/components/dashboard/details-card/DetailItem';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { InfrastructureModel } from '@console/internal/models';
+import { getOCSVersion } from '../../../selectors';
+import { OCSServiceModel } from '../../../models';
+
+const k8sResources: FirehoseResource[] = [
+  {
+    kind: referenceForModel(OCSServiceModel),
+    namespaced: true,
+    isList: false,
+    namespace: 'openshift-storage',
+    prop: 'ocs',
+  },
+  {
+    kind: referenceForModel(SubscriptionModel),
+    namespaced: false,
+    isList: true,
+    prop: 'subscription',
+  },
+];
+
+const DetailsCard = React.memo(
+  withDashboardResources((props: DashboardItemProps) => {
+    const { watchK8sResource, stopWatchK8sResource, resources } = props;
+
+    React.useEffect(() => {
+      k8sResources.forEach((r) => watchK8sResource(r));
+      return () => {
+        k8sResources.forEach((r) => stopWatchK8sResource(r));
+      };
+    }, [watchK8sResource, stopWatchK8sResource]);
+
+    const { ocs, subscription } = resources;
+    const ocsLoaded = _.get(ocs, 'loaded', false);
+    const ocsData = _.get(ocs, 'data') as K8sResourceKind[];
+    const ocsName = getName(_.get(ocsData, 0));
+
+    const subscriptionLoaded = _.get(subscription, 'loaded');
+    const subscriptionVersion = getOCSVersion(subscription as FirehoseResult);
+
+    const [infrastructure, infrastructureLoaded, infrastructureError] = useK8sGet<K8sResourceKind>(
+      InfrastructureModel,
+      'cluster',
+    );
+    const infrastructurePlatform = getInfrastructurePlatform(infrastructure);
+    return (
+      <DashboardCard>
+        <DashboardCardHeader>
+          <DashboardCardTitle>Details</DashboardCardTitle>
+        </DashboardCardHeader>
+        <DashboardCardBody>
+          <DetailItem key="service_name" title="Service Name" isLoading={false} error={false}>
+            OpenShift Container Storage (OCS)
+          </DetailItem>
+          <DetailItem
+            key="cluster_name"
+            title="Cluster Name"
+            error={!ocsLoaded && !_.isEmpty(ocsName)}
```

Comment:
isnt the condition wrong ?
```suggestion
            error={ocsLoaded && !_.isEmpty(ocsName)}
```

you should also add check for `loadError` of `ocs` resource